build.gradle

```diff
@@ -46,6 +47,29 @@ ext.compatibilityVersion = '1.7'
 sourceCompatibility = compatibilityVersion
 targetCompatibility = compatibilityVersion
 
+sourceSets {
+    main.java.srcDirs = ['src/main/java'
+                        ,'src/generated/java' ]
+}
+
+compileJava.doFirst {
+ File generatedDir = file("${rootDir}/src/generated/java")
+ if (generatedDir.exists()) {
+  delete generatedDir
+ }
+ generatedDir.mkdirs()
+}
+
+clean.doFirst {
+ File generatedDir = file("${rootDir}/src/generated/java")
+ if (generatedDir.exists()) {
+  delete generatedDir
+ }
+ generatedDir.mkdirs()
+}
```

Comment:
It can be changed to any folder. Do you have another suggestion on folder name? I may be wrong here, bit this is something I have that is working for me =)